src/dialog-editor/services/modal/modal.ts

```diff
@@ -0,0 +1,298 @@
+   * @param {number} index of field
+   */
+  public showModal(tab: number, box: number, field: number) {
+    let modalOptions = {
+      template: require('./modal.html'),
+      controller: ModalController,
+      controllerAs: 'vm',
+      size: 'lg',
+      resolve: {
+        dialogDetails: function() {
+          return {tabId: tab, boxId: box, fieldId: field};
+        },
+      },
+    };
+    let modal = this.$uibModal.open(modalOptions);
+
+    return modal.result;
+  }
+}
+
+/**
+ * Controller for the Dialog Editor modal service
+ * @ngdoc controller
+ * @name ModalController
+ */
+class ModalController {
+  public modalTab: string = 'element_information';
+  public modalData: any;
+  public element: string;
+  public categories: any;
+  public dialog: any;
+
+  /*@ngInject*/
+  constructor(private dialogDetails: any,
+              private $uibModalInstance: any,
+              private CollectionsApi: any,
+              private DialogEditor: any) {
+    ng.extend(this, {
+      dialog: this.dialogDetails,
+      saveDialogFieldDetails: this.saveDialogFieldDetails,
+      deleteField: this.deleteField,
+      modalUnchanged: this.modalUnchanged,
+      addEntry: this.addEntry,
+      removeEntry: this.removeEntry,
+      modalTabSet: this.modalTabSet,
+      modalTabIsSet: this.modalTabIsSet,
+      currentCategoryEntries: this.currentCategoryEntries,
+    });
+
+    // recognize edited element type
+    if (ng.isUndefined(this.dialog.fieldId)
+     && ng.isUndefined(this.dialog.boxId)
+     && ng.isDefined(this.dialog.tabId)) {
+      this.element = 'tab';
+    } else if (ng.isUndefined(this.dialog.fieldId)
+            && ng.isDefined(this.dialog.boxId)
+            && ng.isDefined(this.dialog.tabId)) {
+      this.element = 'box';
+    } else if (ng.isDefined(this.dialog.fieldId)
+            && ng.isDefined(this.dialog.boxId)
+            && ng.isDefined(this.dialog.tabId)) {
+      this.element = 'field';
+    }
+
+    // clone data from service
+    switch (this.element) {
+      case 'tab':
+        this.modalData = _.cloneDeep(
+          this.DialogEditor.getDialogTabs()[
+            this.DialogEditor.activeTab
+          ]
+        );
+        break;
+      case 'box':
+        this.modalData = _.cloneDeep(
+          this.DialogEditor.getDialogTabs()[
+            this.DialogEditor.activeTab
+          ].dialog_groups[
+            this.dialog.boxId
+          ]
+        );
+        break;
+      case 'field':
+        this.modalData = _.cloneDeep(
+          this.DialogEditor.getDialogTabs()[
+            this.DialogEditor.activeTab
+          ].dialog_groups[
+            this.dialog.boxId
+          ].dialog_fields[
+            this.dialog.fieldId
+          ]
+        );
+        // load categories from API, if the field is Tag Control
+        if (this.modalData.type === 'DialogFieldTagControl') {
+          this.resolveCategories(this.CollectionsApi).then(function(categories: any) {
```

Comment:
`this.resolveCategories()` - resolveCategories doesn't seem to take any arguments